vimiv/utils/files.py

```diff
@@ -26,13 +28,18 @@ def listdir(directory: str, show_hidden: bool = False) -> List[str]:
         directory: Directory to check for files in via os.listdir(directory).
         show_hidden: Include hidden files in output.
     Returns:
-        Sorted list of files in the directory with their absolute path.
+        List of files in the directory with their absolute path.
     """
     directory = os.path.abspath(os.path.expanduser(directory))
+    order_function, reverse = api.settings.image_order.get_para()
     return sorted(
-        os.path.join(directory, path)
-        for path in os.listdir(directory)
-        if show_hidden or not path.startswith(".")
+        [
+            os.path.join(directory, path)
+            for path in os.listdir(directory)
+            if show_hidden or not path.startswith(".")
+        ],
+        key=order_function,
+        reverse=reverse,
```

Comment:
I am currently looking over this, and not quite sure it makes sense to sort by the api setting here. Listdir is essentially used in three places.

1. To get the files in the library. These are later sorted according to directory / image setting anyway.
2. To get the files for the path completion. These should be sorted explicitly, to ensure both image_order and directory_order are respected.
3. To get the files for trash completion. We could use the image_order here, as we don't support deleting directories, but I believe explicitly using it in the completion makes more sense as well.

Thus I would remove the sorting from this all-together. Let me know if I missed something :blush:

Reply:
I guess this has been resolved by 0e4a0db59d977a40638230e1a06e867034ee989f

Reply:
>just wanted to make sure I didn't miss a reason

That is what I though. But do not rely on me too much. I only learned thanks to this review and your commits where else the sorted file paths are also used :laughing: